### PR TITLE
Please Review Class Implementation

### PR DESCRIPTION
**Pull request by ghost**
@all Please review Code 
**OTP Validation Python**

API, Database & Class Implementation Python Project to Validate OTP Real Time Send and Validation & Updates Database for the Same Status.

- Takes Number input from User and Validates the Number based on criteria ie. Number Must be 10 Digit & all integer values.
- then, OTP will generated using random module of python
- then, 'Vonage API' is used to send OTP, if OTP sends successfully return True else False.
- if True, then create a entry for OTP details in Database, and execute validate otp method.
- Validation of OTP is based on following Criteria ie. OTP must be entered within 60 seconds, OTP must be six digit integer value, OTP must be entered in 5 Tries.
- If Validation passed following criteria then Validate OTP and Update status for same as Success.
- Else Throw error Message for same, and update database entry for Failed.

[Screencast from 19-06-24 12:08:59 PM IST.webm](https://github.com/mohit-trootech/OTP-Verification/assets/172255945/cf362b62-674a-4ac7-be59-a4c72a3fabac)

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#
+# client = vonage.Client(key="", secret="wR8CYRRbK7DfiEzL")
+# sms = vonage.Sms(client)
+#
+# responseData = sms.send_message(
+#     {
+#         "from": "Mohit-Trootech",
+#         "to": 917877287800,
+#         "text": 4584,
+#     }
```
Comment (Dhruvi-Makwana) [inline test.py]: Do not push commented code. if it's necessary then ok otherwise not.

File: otp.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+class MobileNumberMisMatch(Exception):
+    """Mobile Number Mismatch Custom Error if Number length is not 10 and number is not numeric"""
+
+    def __init__(self, message: str) -> None:
+        """
+        custom error class to throw mobile Number Mismatch Error
+        @param message: str
+        """
+        self.message = message
```
Comment (Dhruvi-Makwana) [inline otp.py]: from the next time whenever you are write custome exception do not wirte in the same file move to utils. or ask @yatish.prajapati sir either it's right or wrong..

File: test.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# if responseData["messages"][0]["status"] == "0":
+#     print("Message sent successfully.")
+# else:
+#     print(f"Message failed with error: {responseData['messages'][0]['error-text']}")
+import random
+import math
+x = ""
+for i in range(6):
+    x += str(math.floor(random.random() * 10))
+print(x)
```
Comment (Dhruvi-Makwana) [inline test.py]: Give proper variable name.

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,3 @@
 __pycache__
 otp.db
-constant.py
+constant.py
```
Comment (yatishTrootech) [inline .gitignore]: py file should not be in gitignore

**Review by yatishTrootech — APPROVED** [review]

File: otp.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        tries = 5
+        while True:
+            tries -= 1
+            try:
+                if time_difference(datetime.strptime(otp_data[2], "%Y-%m-%d %H:%M:%S")) > 60.00:
+                    raise Exception(constant.EXPIRY_OTP)
+                received_otp = int(input("Enter Received OTP: "))
+                if received_otp == int(otp_data[-2]):
+                    print(constant.VALIDATE_OTP)
+                    self.status_update(otp_data, constant.STATUS_SUCCESS)
```
Comment (yatishTrootech) [inline otp.py:50]: can write this logic in better way

**Review by yatishTrootech — APPROVED** [review]

**Pull request merged**